### PR TITLE
Issues/KUI 1378 fetch data loader

### PR DESCRIPTION
**Pull request by amirhossein-haerian**

### CODE REVIEW

File: public/js/app/components/RoundInformation/RoundInformationInfoGrid.jsx [hunk 1]
```diff
@@ -118,7 +118,7 @@ function RoundInformationInfoGrid({ courseCode, courseRound, selectedSemester })
       </Item>
       <Item title={translation.courseRoundInformation.round_target_group} html={courseRound.round_target_group} />
       <Item title={translation.courseRoundInformation.round_time_slots}>
-        <PlannedModules courseCode={courseCode} courseRound={courseRound} selectedSemester={selectedSemester} />
+        <PlannedModules plannedModules={plannedModules} />
       </Item>
```
Comment (karlandindrakryggen) [inline public/js/app/components/RoundInformation/RoundInformationInfoGrid.jsx:122]: Now when call to `usePlannedModules` is moved up in the tree we can remove PlannedModules.jsx completly and just use the generic html-prop to <Item> here instead, like this:

`<Item title={translation.courseRoundInformation.round_target_group} html={plannedModules} />`

File: public/js/app/components/RoundInformation/PlannedModules.jsx
Comment (karlandindrakryggen) [inline public/js/app/components/RoundInformation/PlannedModules.jsx:1]: Remove this file. See comment on `RoundInformationInfoGrid.jsx`

File: public/js/app/hooks/usePlannedModules.js [hunk 2]
```diff
@@ -12,21 +12,23 @@ export const usePlannedModules = ({ courseCode, semester, applicationCode }) =>
...
-  const { data, isError, setApiParams } = useApi(
-    getPlannedModules,
-    { basePath, courseCode, semester, applicationCode },
-    null,
-    MISSING_INFO
+  const requestData = useMemo(
+    () => ({
+      basePath,
+      courseCode,
+      semester,
```
Comment (karlandindrakryggen) [inline public/js/app/hooks/usePlannedModules.js:19]: I found a bug, the semester prop in the params object is called `selectedSemester` and not `semester`, so semester is always undefined here.

File: public/js/app/hooks/useApi.js [hunk 3]
```diff
@@ -1,20 +1,26 @@
 import { useEffect, useState } from 'react'
 import { STATUS } from './api/status'
 
-export const useApi = (apiToCall, initialApiParams, defaultValue, defaulValueIfNullResponse) => {
-  const [apiParams, setApiParams] = useState(initialApiParams)
+export const useApi = (apiToCall, apiParams, defaultValue, defaulValueIfNullResponse) => {
```
Comment (karlandindrakryggen) [inline public/js/app/hooks/useApi.js:4]: Really like your refactoring here ⭐️

**Review by karlandindrakryggen — CHANGES_REQUESTED** [review]
Looks good! But I found a bug and  cleanup suggestion (remove of PlannedModules file).

**Comment by amirhossein-haerian** [comment]
Thanks for your great comments, I applied the required changes based on them ⭐️

File: public/js/app/hooks/usePlannedModules.js (hunk 2 above)
Comment (karlandindrakryggen) [reply public/js/app/hooks/usePlannedModules.js]: This still doesn't work for when I'm testing...

Comment (karlandindrakryggen) [reply public/js/app/hooks/usePlannedModules.js]: Ah, it is still called semester in the planned modules api function. So it needs to be `semester: selectedSemester,` here.

Like this: 
```
  const requestData = useMemo(
    () => ({
      basePath,
      courseCode,
      semester: selectedSemester,
      applicationCode,
    }),
    [basePath, courseCode, selectedSemester, applicationCode]
  )
```

Comment (amirhossein-haerian) [reply public/js/app/hooks/usePlannedModules.js]: You are absolutely right, I fixed it now; thanks for mentioning this.

**Review by karlandindrakryggen — APPROVED** [review]

**Pull request merged**